### DISCUSSION TITLE
gradlew run without gui for csc file

### DISCUSSION THREAD

**Discussion by ceylanim** [contiki-ng/contiki-ng#2352] [Q&A]
Hi. 
I am running cooja with ./gradlew run. 
I need to run without gui with gradlew. How can I do it? Is there an example?
I need COOJA.testlog. If there is an option to create COOJA.testlog file with gui, I need it too. 
Thank you.

**Comment by pjonsson** [comment]
Things are still changing with the command line options, you can get help for what your Cooja accepts with `./gradlew run --args="--help"`. If you want to pass other options than help, you can pass them within the double quotes.

The `tests` target in `tests/Makefile.simulation-test` in the Contiki-NG repository is updated with how Cooja prefers parameters, disabling gui is the `--no-gui` option. As a general direction, the double dash long parameter names is where things are heading (the command line parsing library, picocli, expects the double dash for long parameter names).

At the time of writing, COOJA.testlog is conditional on running without GUI, so there isn't a way to get that file generated in GUI mode. There are patches in a different tree for (at least partially) changing this behavior, but I don't think those patches will be merged before the end of this year.

If you are prepared to modify your Cooja you should be able to remove the `if (Cooja.isVisualized())`  guards around `logWriter` in `LogScriptEngine.java`.

**Comment by arurke** [comment]
The doc. was expanded in #2376 to cover cooja without gui, see https://docs.contiki-ng.org/en/develop/doc/tutorials/Running-Contiki-NG-in-Cooja.html#running-cooja-without-gui
